devicetypes/widomsrl/widom-smart-dry-contact.src/widom-smart-dry-contact.groovy

```diff
@@ -0,0 +1,339 @@
+	} else if (notSynced) {
+		multiStatusEvent("Sync incomplete! Open settings and tap Done to try again.", true, true)
+	} else {
+		if (device.currentValue("multiStatus")?.contains("Sync")) { multiStatusEvent("Sync OK.", true, true) }
+	}
+}
+
+private multiStatusEvent(String statusValue, boolean force = false, boolean display = false) {
+	if ( !device.currentValue("multiStatus")?.contains("Sync") || device.currentValue("multiStatus") == "Sync OK." || force ) {
+		sendEvent(name: "multiStatus", value: statusValue, descriptionText: statusValue, displayed: display)
+	}
+}
+
+private deviceIdEvent(String value, boolean force = false, boolean display = false) {
+	sendEvent(name: "deviceID", value: value, descriptionText: value, displayed: display)
+}
+
+//event handlers related to configuration and sync
+def zwaveEvent(physicalgraph.zwave.commands.configurationv1.ConfigurationReport cmd) {
+	def paramKey = parameterMap().find( {it.num == cmd.parameterNumber } ).key
+	logging("Parameter ${paramKey} value is ${cmd.scaledConfigurationValue} expected " + state."$paramKey".value, "info")
+	state."$paramKey".state = (state."$paramKey".value == cmd.scaledConfigurationValue) ? "synced" : "incorrect"
+	syncNext()
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.applicationstatusv1.ApplicationRejectedRequest cmd) {
+	logging("rejected request!","warn")
+	for ( param in parameterMap() ) {
+		if (state."$param.key"?.state == "inProgress") {
+			state."$param.key"?.state = "failed"
+			break
+		}
+	}
+}
+//event handlers
+def zwaveEvent(physicalgraph.zwave.commands.basicv1.BasicReport cmd) {
+	//ignore
+}
+def zwaveEvent(physicalgraph.zwave.commands.switchbinaryv1.SwitchBinaryReport cmd) {
+	logging("SwitchBinaryReport received, value: ${cmd.value} ","info")
+	sendEvent([name: "switch", value: (cmd.value == 0 ) ? "off": "on"])
+}
+
+/*
+####################
+## Z-Wave Toolkit ##
+####################
+*/
+def parse(String description) {
+	def result = []
+	def deviceId = [];
+	logging("Parsing: ${description}")
+	if (description.startsWith("Err 106")) {
+	result = createEvent(
+				descriptionText: "Failed to complete the network security key exchange. If you are unable to receive data from it, you must remove it from your network and add it again.",
+				eventType: "ALERT",
+				name: "secureInclusion",
+				value: "failed",
+				displayed: true,
+		)
+	} else if (description == "updated") {
+		return null
+	} else {
+		deviceId = description.split(", ")[0]
+		deviceId = deviceId.split(":")[1]
+		logging("deviceId- ${deviceId}")
+		deviceIdEvent(deviceId, true, true)
+		def cmd = zwave.parse(description, cmdVersions())
+		if (cmd) {
+			logging("Parsed: ${cmd}")
+			zwaveEvent(cmd)
+		}
+	}
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.securityv1.SecurityMessageEncapsulation cmd) {
+	def encapsulatedCommand = cmd.encapsulatedCommand(cmdVersions())
+	if (encapsulatedCommand) {
+    	logging("Parsed SecurityMessageEncapsulation into: ${encapsulatedCommand}")
```

Comment:
Please fix this indentation (4 spaces -> 1 tab)

Reply:
Fixed